chainstate/test-suite/src/tests/orders_tests.rs

```diff
@@ -0,0 +1,664 @@
+            )
+            .add_output(TxOutput::Transfer(
+                OutputValue::TokenV1(token_id, give_amount),
+                Destination::AnyoneCanSpend,
+            ))
+            .build();
+        tf.make_block_builder().add_transaction(tx).build_and_process(&mut rng).unwrap();
+
+        assert_eq!(None, tf.chainstate.get_order_data(&order_id).unwrap());
+        assert_eq!(
+            None,
+            tf.chainstate.get_order_ask_balance(&order_id).unwrap()
+        );
+        assert_eq!(
+            None,
+            tf.chainstate.get_order_give_balance(&order_id).unwrap()
+        );
+    });
+}
+
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn fill_order_check_storage(#[case] seed: Seed) {
+    utils::concurrency::model(move || {
+        let mut rng = make_seedable_rng(seed);
+        let mut tf = TestFramework::builder(&mut rng).build();
+
+        let (token_id, tokens_outpoint, coins_outpoint) = issue_and_mint_token(&mut rng, &mut tf);
+
+        let ask_amount = Amount::from_atoms(rng.gen_range(10u128..1000));
+        let give_amount = Amount::from_atoms(rng.gen_range(10u128..1000));
+        let order_data = OrderData::new(
+            Destination::AnyoneCanSpend,
+            OutputValue::Coin(ask_amount),
+            OutputValue::TokenV1(token_id, give_amount),
+        );
+
+        let order_id = make_order_id(&tokens_outpoint);
+        let tx = TransactionBuilder::new()
+            .add_input(tokens_outpoint.into(), InputWitness::NoSignature(None))
+            .add_output(TxOutput::AnyoneCanTake(order_data.clone()))
+            .build();
+        tf.make_block_builder().add_transaction(tx).build_and_process(&mut rng).unwrap();
+
+        // Fill the order partially
+        let fill_value = OutputValue::Coin(Amount::from_atoms(
+            rng.gen_range(1..ask_amount.into_atoms()),
+        ));
+        let filled_amount = {
+            let db_tx = tf.storage.transaction_ro().unwrap();
+            let orders_db = OrdersAccountingDB::new(&db_tx);
+            orders_accounting::calculate_fill_order(&orders_db, order_id, &fill_value).unwrap()
+        };
+        let left_to_fill = (ask_amount - fill_value.amount()).unwrap();
+
+        let tx = TransactionBuilder::new()
+            .add_input(coins_outpoint.into(), InputWitness::NoSignature(None))
+            .add_input(
+                TxInput::AccountCommand(
+                    AccountNonce::new(0),
+                    AccountCommand::FillOrder(order_id, fill_value, Destination::AnyoneCanSpend),
+                ),
+                InputWitness::NoSignature(None),
+            )
+            .add_output(TxOutput::Transfer(
+                OutputValue::TokenV1(token_id, filled_amount),
+                Destination::AnyoneCanSpend,
+            ))
+            .add_output(TxOutput::Transfer(
+                OutputValue::Coin(left_to_fill),
+                Destination::AnyoneCanSpend,
+            ))
```

Comment:
This output is confusing... so I spent some amount to fill an order, and got the corresponding token in the first output. But then why does this output exist here, one that contains the amount left in the order?

Reply:
this is utxo for taker to fill the order completely in a later tx

Reply:
Please explain it in comments. Explain these steps.

Reply:
I would prefer to use a different number/variable to make this behavior clear.

Reply:
Well, thinking about it, why do we have to include this in this transaction? I'd rather not have it here to have the intent clear.

Reply:
the idea was to fill partially and then fill the order completely. For that I carry the change across txs and made it precisely the same to hit that case.